types/shares_test.go

```diff
@@ -74,8 +75,8 @@ func TestMakeShares(t *testing.T) {
 			NamespacedShares{
 				NamespacedShare{
 					Share: append(
-						append(reservedTxNamespaceID, byte(0)),
-						zeroPadIfNecessary(smolTxLenDelimited, TxShareSize)...,
+						reservedTxNamespaceID,
+						zeroPadIfNecessary(smolTxLenDelimited, MsgShareSize)...,
```

Comment:
I was confused about the `*` byte mentioned in the spec. I didn't know that it always had to be there. Please, ignore, I'll change this later.

Reply:
That is why I insisted on pseudo-code (or you know, an actual implementation) instead of that diagram in the spec.